core/modules/qproc/QuerySession.cc

```diff
@@ -356,21 +354,25 @@ void QuerySession::_applyConcretePlugins() {
 }
 
 /// Some code useful for debugging.
-void QuerySession::_showFinal(std::ostream& os) {
-    // Print out the end result.
-    query::QueryTemplate par = _stmtParallel.front()->getTemplate();
-    query::QueryTemplate mer = _stmtMerge->getTemplate();
-
-    os << "\tQuerySession::_showFinal() : parallel: " << par.dbgStr() << std::endl;
-    os << "\tQuerySession::_showFinal() : merge: " << mer.dbgStr() << std::endl;
+std::string QuerySession::toString() {
```

Comment:
I agree we should standardize how to serialize C++ objects in Qserv, but this is not the goal of this ticket, isn'it? Here I've only renamed the method.  Maybe we should open a ticket which will explain how to define this output tools. Please note that lsst/log has to use a std::string toString() function, so `operator <<` is not require for logging